File: src/python/11_napisy_wprowadzenie/zad12.py
"""
ZAD-12 — Usuń spacje ze zdania

**Poziom:** ★☆☆
**Tagi:** `replace`, `string`

### Treść

Wczytaj zdanie i usuń z niego wszystkie spacje.

### Wejście

* 1. linia: zdanie

### Wyjście

* 1. linia: zdanie bez spacji

### Przykład

**Wejście:**

```
Ala ma kota
```

**Wyjście:**

```
Alamakota
```

"""


def usun_spacje(zdanie):
    return zdanie.replace(" ", "")


def test_usun_spacje():
    assert usun_spacje("Ala ma kota") == "Alamakota"


if __name__ == "__main__":

    test_usun_spacje()

File: src/haskell/09_listy_wprowadzenie/zad12.hs
{-
ZAD-12 — Rotacja w lewo / prawo

\**Poziom:** ★★☆
\**Tagi:** `listy`, `rotacja`, `modulo`

### Treść

Wczytaj listę liczb całkowitych, a następnie:

\* `kierunek = 0` → rotacja w lewo,
\* `kierunek = 1` → rotacja w prawo,

o `K` pozycji (gdzie `K ≥ 0`). Wypisz listę po rotacji, przecinkami bez spacji.

### Wejście

\* 1. linia: `N` (`N ≥ 1`)
\* kolejne `N` linii: liczby całkowite
\* kolejna linia: `kierunek` (0 lub 1)
\* ostatnia linia: `K` (`K ≥ 0`)

### Wyjście

Jedna linia: lista po rotacji, elementy oddzielone przecinkami.

### Przykład

\**Wejście:**

```
7
5
27
6
2
1
10
8
0
2
```

\**Wyjście:**

```
6,2,1,10,8,5,27
```

### Uwagi

\* Zredukuj `K` przez `K % N`.

-}
main :: IO ()
main = do
  n <- readLn :: IO Int
  nums <- mapM (\_ -> readLn :: IO Int) [1 .. n]
  kierunek <- readLn :: IO Int
  k <- readLn :: IO Int

  let intercalate sep [] = ""
      intercalate sep [x] = x
      intercalate sep (x : xs) = x ++ sep ++ intercalate sep xs
  let k' = k `mod` n
  let result =
        if kierunek == 0
          then drop k' nums ++ take k' nums -- lewo
          else drop (n - k') nums ++ take (n - k') nums -- prawo
  putStrLn $ intercalate "," $ map show result
